inventory/host_vars/network.yml

```
@@ -45,3 +45,12 @@ yamllint:
         /tests/tests_bond_cloned_mac_initscripts.yml
         /tests/tests_bridge_cloned_mac_initscripts.yml
         /tests/tests_bridge_cloned_mac_nm.yml
+# these are added to the list of default status check contexts
+status_check_contexts:
+  - DCO
+  - markdownlint
+  - integration (c8s)
+  - integration (c9s)
+  - integration (c7)
+  - python-26
+  - woke
```

```suggestion
  - DCO
  - markdownlint
  - integration (c8s)
  - integration (c9s)
  - integration (c7)
  - python-26
  - woke
  - ansible_lint
  - ansible_plugin_scan
  - ansbible_test
  - Analyze (python)
  - shellcheck
  - ansible_managed_var_comment
```

Currently, the UI also shows a source for each check (DCO, GitHub Actions, GitHub Code Scanning) depending on the status, the alternative seems to be "any source" which might be what this will be configuring, since it does not collect the source. It would be great to keep the sources as well.

---

The `ansible_lint` etc. are the defaults applied to all roles - so the network.yml contains only those status checks unique (currently) to network

---

> Currently, the UI also shows a source for each check (DCO, GitHub Actions, GitHub Code Scanning) depending on the status, the alternative seems to be "any source" which might be what this will be configuring, since it does not collect the source. It would be great to keep the sources as well.

Why keep the sources?

---

> > Currently, the UI also shows a source for each check (DCO, GitHub Actions, GitHub Code Scanning) depending on the status, the alternative seems to be "any source" which might be what this will be configuring, since it does not collect the source. It would be great to keep the sources as well.
> 
> Why keep the sources?

My assumption is that there is that there might be a security impact (maybe not that big since it seems to require write access to the repo to set a state) and to avoid that status checks conflict. For example, if someone experiments with a new tool that would set a markdownlint status, it could approve the PR without the official markdownlint being run. I guess, this also would allow users to circumvent the branch protection rules but they need to have already write access.

---

Here is the API that allows you to create a status check context with the id: https://docs.github.com/en/graphql/reference/input-objects#requiredstatuscheckinput - the `appId` field is used to specify a particular action that must already be in the database.  Here is the output of searching the network role with the full requiredStatusCheck name + app information:
```
                createdAt: '2017-03-25T13:51:49Z'
                databaseId: 1861
                description: "This App enforces the [Developer Certificate of Origin](https://developercertificate.org/)\
                    \ (DCO) on Pull Requests. It requires all commit messages to contain\
                ...
                url: https://probot.github.io/apps/dco/
            context: DCO
...
                createdAt: '2020-03-17T21:25:09Z'
                databaseId: 57789
                description: ''
                name: GitHub Code Scanning
                slug: github-code-scanning
                updatedAt: '2021-12-14T16:45:01Z'
                url: https://github.com/features/security
            context: CodeQL
...
                createdAt: '2018-07-30T09:30:17Z'
                databaseId: 15368
                description: Automate your workflow from idea to production
                slug: github-actions
                updatedAt: '2019-12-10T19:04:12Z'
                url: https://help.github.com/en/actions
            context: markdownlint
....
```
Except for `DCO` and `CodeQL`, every "app" is the generic github-actions "app".  I guess I could change the input to reflect that. I'm assuming the `databaseId` field in the output is the `appId` field I should use in the input - the documentation doesn't really say.  It's odd that "real" github actions like ansible_lint do not have their own appId - perhaps because we have custom code in there - for the `ansible-lint` check we have some custom code to munge meta/main.yml role_name and namespace - for `ansible-test` we have code to convert to collection.

Note that the key is the `context` - this must exactly match the name of the `job` e.g. https://github.com/linux-system-roles/.github/blob/main/playbooks/templates/.github/workflows/ansible-lint.yml#L12
```yaml
jobs:
  ansible_lint:
```
so as long as there is only 1 github action which defines `jobs.ansible_lint`, we can be guaranteed that if we refer to a status check context name as `ansible_lint` it will use the one defined in .github/workflows/ansible-lint.yml, which requires write access to the repo to change.

Unless you can foresee some attack vector based on status check context naming, I would prefer to use status check context name as is being used currently in this PR.